### PR TITLE
Make platform exclusions

### PR DESCRIPTION
**Pull request by dimitre**
Testing a way of setting platform exclusions per project
so a specific project can be compiled without many parts of OF, example: 3d, video, graphics
or libraries, like boost, poco, etc.

I'm testing some configurations in projectGenerator config.make to see how much of OF it can exclude
it is a program who doesn't need most of the things from core

### CODE REVIEW

File: libs/openFrameworks/app/ofAppRunner.cpp
Comment (dimitre) [inline libs/openFrameworks/app/ofAppRunner.cpp:1]: I've removed ofImage and ofTruetype here, they were only used to shutdown truetype and freeimage, and they can be declared as forward declarations as ofURLFileLoaderShutdown already is.
I see no disadvantage of doing this and the advantage is we can build projects without Freeimage and Freetype when needed.

File: libs/openFrameworksCompiled/project/makefileCommon/config.shared.mk
Comment (dimitre) [inline libs/openFrameworksCompiled/project/makefileCommon/config.shared.mk:1]: now ALL_OF_CORE_SOURCE_PATHS can be filtered by CORE_EXCLUSIONS

File: .github/workflows/build-macos.yml
Comment (dimitre) [inline .github/workflows/build-macos.yml:1]: not meant to be in this PR but it would be great to cache this path too.

**Comment by dimitre** [comment]
I've now extended the possibility to other platforms also, 
if you have the time please take a look, some proofreading @ofTheo or @NickHardeman

**Comment by Jonathhhan** [comment]
> I'm testing some configurations in projectGenerator config.make to see how much of OF it can exclude
> it is a program who doesn't need most of the things from core

would it work the same way with `make` and linux and without the project generator (I have no idea)?

**Comment by dimitre** [comment]
@Jonathhhan sorry I dont' fully understand your question.

File: libs/openFrameworks/app/ofAppRunner.cpp
Comment (ofTheo) [reply libs/openFrameworks/app/ofAppRunner.cpp:1]: @dimitre - not sure I totally understand.  
How is:  `ofTrueTypeShutdown()` getting called now if this PR is merged?

Comment (dimitre) [reply libs/openFrameworks/app/ofAppRunner.cpp:1]: yes this is invoked from ofTrueTypeFont.cpp, it is overridden. anyway there is no code inside this function anymore

Comment (dimitre) [reply libs/openFrameworks/app/ofAppRunner.cpp:1]: You can check the way ofURLFileLoaderShutdown is being utilized, that has a forward declarator there, and then it was overridden with the proper function on ofURLFileLoader.cpp

Comment (ofTheo) [reply libs/openFrameworks/app/ofAppRunner.cpp:1]: Ah I see - yeah the function does nothing. 🙂
And `ofCloseFreeImage` is still getting called in the exit callback anyway. Cool!

Comment (dimitre) [reply libs/openFrameworks/app/ofAppRunner.cpp:1]: Yes and the nice thing it allows us to build OF without truetypefont when needed too.

**Pull request merged**